pkg/operation/botanist/namespaces.go

```diff
@@ -103,3 +116,31 @@ func (b *Botanist) WaitUntilSeedNamespaceDeleted(ctx context.Context) error {
 func (b *Botanist) DefaultShootNamespaces() component.DeployWaiter {
 	return namespaces.New(b.K8sSeedClient.Client(), b.Shoot.SeedNamespace)
 }
+
+// getShootExtensionTypes returns all extension types that are enabled or explicitly disabled for the shoot.
+func getShootExtensionTypes(b *Botanist, ctx context.Context) (map[string]bool, error) {
```

Comment:
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx: 
ref: https://pkg.go.dev/context@go1.16.7

```suggestion
func getShootExtensionTypes(ctx context.Context, b *Botanist) (map[string]bool, error) {
```